tests/php/src/Cli/Export/ExportWidgets.php

```diff
@@ -0,0 +1,39 @@
+<?php
+/**
+ * Reference site export widgets.
+ *
+ * @package AmpProject\AmpWP
+ */
+
+namespace AmpProject\AmpWP\Tests\Cli\Export;
+
+use AmpProject\AmpWP\Tests\Cli\ExportStep;
+
+final class ExportWidgets implements ExportStep {
+
+	/**
+	 * Process the export step.
+	 *
+	 * @param ExportResult $export_result Export result to adapt.
+	 *
+	 * @return ExportResult Adapted export result.
+	 */
+	public function process( ExportResult $export_result ) {
+		$exported_widgets = [];
+
+		foreach ( wp_get_sidebars_widgets() as $sidebar => $widgets ) {
+			$exported_widgets[ $sidebar ] = [];
+			foreach ( $widgets as $widget_instance_id ) {
+				// Get id_base (remove -# from end) and instance ID number.
+				$id_base = preg_replace( '/-[0-9]+$/', '', $widget_instance_id );
+				$instance = preg_replace( '/^' . preg_quote( $id_base, '/' ) . '-/', '', $widget_instance_id );
+				$widget_options = get_option( 'widget_' . $id_base, [] );
+				$exported_widgets[ $sidebar ][ $widget_instance_id ] = $widget_options[ (int) $instance ];
```

Comment:
While this will work for most sites now, the option name used to store a `WP_Widget`'s settings is not guaranteed to include the `$id_base`. The actual option name is stored in `WP_Widget::$option_name`. There is also a `\WP_Widget::$alt_option_name`! Lastly, widgets may not use `WP_Widget` at all. For example: https://gist.github.com/westonruter/7141599

Reply:
Yes, this was mainly a shortcut based what I assumed would be what we'd encounter while exporting the sites we'd just build.

I'll revisit this exporter to provide more complete support.

Reply:
Probably only need to address it when it becomes an issue.

Reply:
That was the initial plan. 😀
Let's stick to that than!